components/core/src/clp_s/JsonConstructor.cpp

```diff
@@ -32,6 +32,15 @@ JsonConstructor::JsonConstructor(JsonConstructorOption const& option) : m_option
         );
     }
 
+    if (false == std::filesystem::exists(option.temp_output_dir)) {
```

Comment:
We only want to perform this check if temp_output_dir is being used right?

Also, do we want to try creating this temporary directory if it doesn't exist? Only asking because I'm not sure what the intended behaviour is.

Reply:
Ah for my first point I see that it works as is because we set m_temp_output_dir = m_output_dir when the argument isn't being used. Still somewhat confusing as written though.

I know its more code but we should maybe consider wrapping temp_output_dir an std::optional throughout the code base to make it a bit more explicit whether its being used or not.